### PR TITLE
Updated Runecraft Plugin

### PR DESCRIPTION
**Pull request by lulwut**
- Lava Rune Changes:
 - Added many menu swaps to make the process entirely left click
- Overlay added notifying broken pouch
- Issue: Easyscape has to be turned off or conflicts turned off to work with lava rune menuentry changes.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 1]
```diff
@@ -26,64 +26,58 @@
...
-import net.runelite.api.events.ChatMessage;
-import net.runelite.api.events.ConfigChanged;
-import net.runelite.api.events.DecorativeObjectDespawned;
-import net.runelite.api.events.DecorativeObjectSpawned;
-import net.runelite.api.events.GameStateChanged;
-import net.runelite.api.events.ItemContainerChanged;
-import net.runelite.api.events.MenuEntryAdded;
-import net.runelite.api.events.NpcDespawned;
-import net.runelite.api.events.NpcSpawned;
+import net.runelite.api.*;
```
Comment (Ganom) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Don't use wildcard imports.

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssRifts.java [hunk 2]
```diff
@@ -64,18 +64,14 @@
 	@Getter
 	private final int itemId;
 
-	private static final Map<Integer, AbyssRifts> rifts;
+	private static final Map<Integer, AbyssRifts> rifts = new HashMap<>();
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssRifts.java]: Why did you change this?

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftOverlay.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package net.runelite.client.plugins.runecraft;
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftOverlay.java]: I feel like this would be better as a widgetitemoverlay, which only gets added when a pouch is degraded

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 4]
```diff
@@ -103,121 +97,118 @@
...
 	@Inject
 	private RunecraftConfig config;
 
 	@Inject
 	private Notifier notifier;
 
 	@Provides
-	RunecraftConfig getConfig(ConfigManager configManager)
-	{
+	RunecraftConfig getConfig(ConfigManager configManager) {
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: checkstyle, { should be on a new line

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 5]
```diff
@@ -103,121 +97,118 @@
...
-		{
+	public void onMenuEntryAdded(MenuEntryAdded entry) {
+		if (client.getGameState() != GameState.LOGGED_IN) {
+			return;
+		}
+
+		Widget loginScreenOne = client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN);
+		Widget loginScreenTwo = client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN_MESSAGE_OF_THE_DAY);
+
+		if (loginScreenOne != null || loginScreenTwo != null) {
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: everything  above thsi isnt really needed, also why remove the wearing tiara switch? now it's always on and not checking for config or anything

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 6]
```diff
@@ -103,121 +97,118 @@
...
-				swap(client, "empty", option, target);
-			}
-		}
 
-		else if (target.contains("ring of dueling") && option.contains("remove"))
-		{
-			if (target.contains("7") || target.contains("5") || target.contains("3") || target.contains("1"))
-			{
-				swap(client, "castle wars", option, target);
+		Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: the way I did this (with deposit-all/type/id) made sure it only worked in bank as well, this is a unneeded  change

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 7]
```diff
@@ -103,121 +97,118 @@
...
+		if (isEssencePouch(target)) {
+			if (widgetBankTitleBar == null || widgetBankTitleBar.isHidden()) { //swap pouches based on empty/full and in bank
+				swap(client, "Empty", option, target);
+			} else {
+				swap(client, "Fill", option, target); //Due to RuneLite issues the "Deposit" menutext will always show even though it is on fill
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: no runelite issues, jagex issues

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 8]
```diff
@@ -103,121 +97,118 @@
...
 			}
-			else if (wearingBindingNeck())
-			{
+		} else if (target.contains("ring of dueling") && option.contains("withdraw")) { //withdraw-1 ring of dueling
+			swap(client, "withdraw-1", option, target);
+		} else if (target.contains("binding necklace") && option.contains("withdraw")) { //withdraw-1 binding necklace
+			swap(client, "withdraw-1", option, target);
+		}
+		else if (target.contains("ring of dueling") && option.contains("remove")) {
+			if (client.getLocalPlayer().getWorldLocation().getRegionID() != 10315) { //changes duel ring teleport options based on location
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: instead of checking if you're at the fire altar, check if you're at castle wars?

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 9]
```diff
@@ -103,121 +97,118 @@
...
+				swap(client, "Teleport", option, target);
+			} else if(target.contains("max cape") && option.contains("remove")) { //teleport for max cape
+				swap(client, "Crafting Guild", option, target);
+			}
+		 else if (target.contains("altar") && option.contains("craft")) // Don't accidentally click the altar to craft
 		{
 			hide(option, target, true);
-		}
-		else if (target.contains("pure") && option.contains("use")) // Don't accidentally use pure essence on altar
+		} else if (target.contains("pure")) // Don't accidentally use pure essence on altar
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: why change this, you don't have to run the hide code on pure essence in bank/wherever

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java [hunk 10]
```diff
@@ -228,51 +219,44 @@ private void hide(String option, String target, boolean contains)
...
+					&& (entryTarget.equals(target) || (entryTarget.contains(target) && contains))) {
 				return i;
 			}
 		}
 
 		return -1;
 	}
 
+
+	private boolean isEssencePouch(String target) {
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: make pouches a immutable set and use Set.contains()

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java (hunk 7 above)
Comment (lulwut) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Yeah sorry, jagex*

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java (hunk 8 above)
Comment (lulwut) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: People can bank at other places, I bank at crafting guild

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java (hunk 9 above)
Comment (lulwut) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Oops, didn't mean to

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssRifts.java (hunk 2 above)
Comment (lulwut) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssRifts.java]: No clue, I didn't even touch anything in the Abyss stuff o.o

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java (hunk 5 above)
Comment (lulwut) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: I figured if the plugin config option was switched on then it would be ok? I can see why the tiara/cape thing is a better idea.

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java (hunk 8 above)
Comment (wetrag) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java]: Can we get this swap for Stamina (1) and Binding necklaces :)
